src/test/java/com/endava/internship/collections/StudentSetTest.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    @Test
+    void testClear() {
+        studentSetFilling();
+
+        studentSetTest.clear();
+
+        assertTrue(studentSetTest.isEmpty(), "After cleaning, the collection thinks it is not empty");
+        assertEquals(0, studentSetTest.size(), "After cleaning, the collection size is wrong");
+    }
+
+    @Test
+    void testAddAll() {
+        studentSetFilling();
+
+        final List<Student> students = new ArrayList<>();
+        students.add(student3);
+        students.add(student4);
+
+        studentSetTest.addAll(students);
+
+        assertTrue(studentSetTest.contains(student3), "The collection does not contains items added by the addAll method");
+        assertTrue(studentSetTest.contains(student4), "The collection does not contains items added by the addAll method");
+    }
+
+    @Test
+    void testContainsAll() {
+        studentSetFilling();
+
+        final List<Student> studentList = new ArrayList<>();
+        studentList.add(student2);
+        studentList.add(student1);
+
+        assertTrue(studentSetTest.containsAll(studentList), "The collection does not contain required elements");
+    }
+
+    @Test
+    void testContainsAllWhenUnsupportedOperation() {
+        studentSetFilling();
+
+        final List<Object> objectsList = new ArrayList<>();
+        objectsList.add(new Object());
+        objectsList.add(new Object());
+
+        assertThrows(UnsupportedOperationException.class, () -> studentSetTest.containsAll(objectsList));
+    }
+
+    @Test
+    void testRemoveAllWhenUnsupportedOperation() {
+        studentSetFilling();
+
+        final List<Object> objectsList = new ArrayList<>();
+        objectsList.add(new Object());
+        objectsList.add(new Object());
+
+        assertThrows(UnsupportedOperationException.class, () -> studentSetTest.removeAll(objectsList));
+    }
+
+    @Test
+    void testRetainAll() {
+        studentSetFilling();
+
+        final List<Student> students = new ArrayList<>();
+        students.add(student2);
+        students.add(student3);
+
+        studentSetTest.retainAll(students);
+        assertEquals(1, studentSetTest.size(), "After executing the method, the size of the collection is incorrect");
+        assertTrue(studentSetTest.contains(student2), "After executing the method, the collection does not contain the required elements");
+    }
+
+    @Test
+    void testRemoveAll() {
+        studentSetFilling();
+        studentSetTest.add(student3);
+        studentSetTest.add(student4);
+
+        final List<Student> students = new ArrayList<>();
+        students.add(student2);
+        students.add(student4);
+
+        studentSetTest.removeAll(students);
+
+        assertFalse(studentSetTest.containsAll(students), "The removeAll method did not remove the item from collection");
+        assertEquals(2, studentSetTest.size(), "After executing the method, the size of the collection is incorrect");
+    }
+
+    private void studentSetFilling() {
```

Comment:
you can remake this method to receive integer as argument and fill you studentSet with that number of students(and use this method in your tests when you need to fill your studentSet)